libs/wingsdk/src/target-tf-aws/redis.ts

```diff
@@ -0,0 +1,135 @@
+import { ElasticacheCluster } from "@cdktf/provider-aws/lib/elasticache-cluster";
+import { ElasticacheSubnetGroup } from "@cdktf/provider-aws/lib/elasticache-subnet-group";
+import { SecurityGroup } from "@cdktf/provider-aws/lib/security-group";
+import { Subnet } from "@cdktf/provider-aws/lib/subnet";
+import { Construct } from "constructs";
+import { App } from "./app";
+import { Function } from "./function";
+import { Code } from "../core";
+import * as core from "../core";
+import * as redis from "../redis";
+import { NameOptions, ResourceNames } from "../utils/resource-names";
+
+const ELASTICACHE_NAME_OPTS: NameOptions = {
+  maxLen: 50,
+  disallowedRegex: /([^a-z0-9]+)/g,
```

Comment:
I'm guessing we can allow upper case characters here, as long as we apply the right case conversion:
```suggestion
  disallowedRegex: /([^a-zA-Z0-9]+)/g,
  case: CaseConventions.LOWERCASE,
```

(this API still needs some work I think)